pkg/snmp/gosnmplib/walk.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package gosnmplib
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gosnmp/gosnmp"
+)
+
+const (
+	// Base OID for MIB-2 defined SNMP variables
+	// This is the lowest OID that anything should return (most will either be under this prefix or under 1.3.6.1.4.1)
+	baseOID = ".1.3.6.1.2.1"
```

Comment:
LLDP data is found under `.1.0.*`, I think this would be good to have in a device scan.
Wondering if baseOID should just be `.` so that we really walk the whole thing, wdyt?